File: python_snacks/python/mortgagecalculator.py


principal_amount= int(input('Enter the principal-amount:'))
annual_interest_rate= int(input('Enter the annual interest rate:'))
duration_in_years = int(input('Enter the duration in years:'))
p = principal_amount
r = annual_interest_rate / 100 /12 
n = duration_in_years * 12


monthly_payment= p * (r* (1+r)**n) / (((1+r)**n)-1)
print(monthly_payment)

File: java/mortgagecalculator.java

import java.util.Scanner;

public class MortgageCalculator {  

public static void main(String[]args) {
Scanner input = new Scanner(System.in);

	System.out.print("Enter the principal amount:");
	int p = input.nextInt();

	System.out.print("Enter the annual interest rate:");
	int r = input.nextInt();

	System.out.print("Enter the duration in years:");
	int n = input.nextInt();

	p = principal_amount;
	r = annual_interest_rate / 100 /12; 
	n = duration_in_years * 12;

	int monthlyPayment = p * (r* (1+r)* n) / (((1+r)*n)-1);

	System.out.printf("%d%n",monthlyPayment);


	}

}


	
